tests/integration/targets/ndo_ptp_policy/tasks/main.yml

```diff
@@ -0,0 +1,233 @@
+        global_priority1: 100
+        global_priority2: 200
+        state: present
+      check_mode: true
+      register: cm_update_ptp_policy
+
+    - name: Update a PTP policy
+      cisco.mso.ndo_ptp_policy:
+        <<: *update_ptp_policy
+      register: nm_update_ptp_policy
+
+    - name: Update a PTP policy again
+      cisco.mso.ndo_ptp_policy:
+        <<: *update_ptp_policy
+      register: nm_update_ptp_policy_again
+
+    - name: Assert PTP policy was updated
+      assert:
+        that:
+          - cm_update_ptp_policy is changed
+          - cm_update_ptp_policy.previous.description == ""
+          - cm_update_ptp_policy.previous.global.adminState == "enabled"
+          - cm_update_ptp_policy.previous.global.fabSyncIntvl == -3
+          - cm_update_ptp_policy.current == cm_update_ptp_policy.proposed
+          - cm_update_ptp_policy.current.description == cm_update_ptp_policy.proposed.description == "changed_description"
+          - cm_update_ptp_policy.current.global.adminState == cm_update_ptp_policy.proposed.global.adminState == "disabled"
+          - cm_update_ptp_policy.current.global.fabSyncIntvl == cm_update_ptp_policy.proposed.global.fabSyncIntvl == -2
+          - cm_update_ptp_policy.current.global.prio1 == cm_update_ptp_policy.proposed.global.prio1 == 100
+          - cm_update_ptp_policy.current.global.prio2 == cm_update_ptp_policy.proposed.global.prio2 == 200
+          - cm_update_ptp_policy.current.global.globalDomain == cm_update_ptp_policy.proposed.global.globalDomain == 1
+          - nm_update_ptp_policy is changed
+          - nm_update_ptp_policy.current.description == "changed_description"
+          - nm_update_ptp_policy.current.global.adminState == "disabled"
+          - nm_update_ptp_policy.current.global.fabSyncIntvl == -2
+          - nm_update_ptp_policy.current.global.globalDomain == 1
+          - nm_update_ptp_policy.current.global.prio1 == 100
+          - nm_update_ptp_policy.current.global.prio2 == 200
+          - nm_update_ptp_policy_again is not changed
+
+    # QUERY
+    - name: Update PTP policy name with UUID
+      cisco.mso.ndo_ptp_policy:
+        <<: *update_ptp_policy
+        ptp_policy: ansible_test_ptp_policy_updated
+        ptp_policy_uuid: "{{ nm_add_ptp_policy.current.uuid }}"
+        state: present
+
+    - name: Query a PTP policy with ptp_policy name
+      cisco.mso.ndo_ptp_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        ptp_policy: ansible_test_ptp_policy_updated
+        state: query
+      register: query_one
+
+    - name: Query all PTP policies in the template
+      cisco.mso.ndo_ptp_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        state: query
+      register: query_all
+
+    - name: Verify query_one and query_all
+      assert:
+        that:
+          - query_all is not changed
+          - query_one is not changed
+          - query_all.current == query_one.current
```

Comment:
Should we add another PTP policy to test querying more than 1 policy?

Reply:
Only one PTP policy can be added